app/graph/types/annotation_object.rb

```diff
@@ -0,0 +1,82 @@
+class AnnotationObject < BaseObject
+  class << self
+    def define_shared_behavior(subclass, mutation_target)
+      subclass.class_variable_set(:@@mutation_target, mutation_target)
```

Comment:
TODO: See if it makes sense to update this to be more similar to how we do mutations, rather than using a class variable

Reply:
it's not easily bc we need access to module constants before class is loaded